### ISSUE TITLE
Crashes with core dump and tests fail since upgrading json-c to v0.13

### ISSUE THREAD

**Issue by maximbaz** [Yubico/libu2f-server#32]
`json-c` v0.13 has been released and it was published to Arch Linux repositories today, and thus this project stopped working for everyone.

------

<details>
<summary>First of all, the tests started failing - click to see the output of <code>$ make check</code></summary>

```
Making check in u2f-server
make[1]: Entering directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/u2f-server'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/u2f-server'
Making check in src
make[1]: Entering directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/src'
make  check-am
make[2]: Entering directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/src'
make[2]: Nothing to be done for 'check-am'.
make[2]: Leaving directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/src'
make[1]: Leaving directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/src'
Making check in tests
make[1]: Entering directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
make  basic core openssl \
  u2f-server-test.sh
make[2]: Entering directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
  CC       basic.o
  CCLD     basic
  CC       core.o
  CCLD     core
  CC       openssl.o
  CCLD     openssl
make[2]: Nothing to be done for 'u2f-server-test.sh'.
make[2]: Leaving directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
make  check-TESTS
make[2]: Entering directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
make[3]: Entering directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
PASS: basic
PASS: core
PASS: openssl
ERROR: u2f-server-test.sh
make[4]: Entering directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
============================================================================
Testsuite summary for libu2f-server 1.0.1
============================================================================
# TOTAL: 4
# PASS:  3
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 1
============================================================================
See tests/test-suite.log
Please report to yubico-devel@googlegroups.com
============================================================================
make[3]: *** [Makefile:725: test-suite.log] Error 1
make[3]: Leaving directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
make[2]: *** [Makefile:833: check-TESTS] Error 2
make[2]: Leaving directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
make[1]: *** [Makefile:918: check-am] Error 2
make[1]: Leaving directory '/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/tests'
make: *** [Makefile:430: check-recursive] Error 1
```
</details>

-------

<details>
<summary>Click to see the contents of <code>tests/test-suite.log</code></summary>

```
===============================================
   libu2f-server 1.0.1: tests/test-suite.log
===============================================

# TOTAL: 4
# PASS:  3
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 1

.. contents:: :depth: 2

ERROR: u2f-server-test
======================

/tmp/makepkg/libu2f-server/src/libu2f-server-1.0.1/src/.libs/lt-u2f-server: unrecognized option '--unknown'
error: An origin must be specified with -o
error: An appId must be specified with -i
error: origin must be http, https or pam
error: (-8) Format mismatch
warning: challenge should be 43 characters long
```
</details>

------------

If I ignore the test failures and install it anyway, everything will be crashing in runtime, for example if I recompile `pam_u2f` and try to use it. For example, running `pamu2fcfg -utest` produces the following:

```
❯  pamu2fcfg -utest
[1]    9028 segmentation fault (core dumped)  pamu2fcfg -utest

❯  pamu2fcfg -utest
*** Error in `pamu2fcfg': corrupted double-linked list: 0x000055be7d59d2c0 ***
[1]    9087 abort (core dumped)  pamu2fcfg -utest
```

Downgrading `json-c` is not really an option, as there are other apps that have already been recompiled in Arch Linux repository against the latest `json-c`. 

Please have a look since everyone on Arch Linux is affected by this issue.

**UPDATE**: The current `master` doesn't have the issue, please prioritize releasing a new version #27 so that we can upgrade the package on Arch Linux.

**Comment by fmauNeko** [comment]
I just released libu2f-server-git on AUR as a temporary fix

**Comment by a-dma** [comment]
I just released `1.1.0`

**Comment by maximbaz** [comment]
It is perfect now, will deal with updating AUR package separately - thanks for quick fix!
